src/index.ts

```diff
@@ -20,6 +20,7 @@ export type StateCreator<T extends State> = (
 ) => T
 export type SetState<T extends State> = (partial: PartialState<T>) => void
 export type GetState<T extends State> = () => T
+export type ReturnState<T extends State, U> = U extends T ? T : U
```

Comment:
No need for `ReturnState` because the return value for `useStore()` and `useStore(selector)` would be declared in the interface.

Reply:
Done